arch/xtensa/include/xtensa_asm2_s.h

```diff
@@ -184,6 +186,20 @@
 #if XCHAL_HAVE_FP && defined(CONFIG_CPU_HAS_FPU) && defined(CONFIG_FPU_SHARING)
 	FPU_REG_SAVE
 #endif
+
+#if (XCHAL_CP_MASK & (1 << XCHAL_CP_ID_AUDIOENGINELX)) && \
+	defined(CONFIG_CPU_HAS_DSP) && defined(CONFIG_DSP_SHARING)
+
+	/* Put aligned address of ae_regs into A2 */
+
+	addi a2, a1, ___xtensa_irq_bsa_t_ae_regs_OFFSET
+	movi a3, ~(XCHAL_CP1_SA_ALIGN - 1)
+	addi a2, a2, (XCHAL_CP1_SA_ALIGN - 1)
+	and a2, a2, a3
+
+	xchal_cp1_store a2 a0 a3 a3 a3  /* Only A0 and A2 are used by macro */
+#endif
```

Comment:
This macro says that a2/a3 won't be modified at exit. I didn't see any place assuming it, but it is documented and need to be confirmed and then, at least change the documentation.